### DISCUSSION TITLE
How to mapping ordinal_encode result value and original value?

### DISCUSSION THREAD

**Discussion by tommyhj217** [vaexio/vaex#1927] [Q&A] [answered]
Hi 

I'm using it because I'm interested in vaex.

How can the Int value be mapped to the existing string value after the original_encode?

df = df.ordinal_encode(columns="test")

The value will change int after this command, and if there is a way to know which value is the enum value, please share it.

Thanks
Hyunjun

**Comment by JovanVeljanoski** [comment, marked as answer]
Hi, 

For now you can do this via 
`df._categories`. They are sorted by the way they appear in that dictionary, starting from 0.

In the future major version of vaex (i.e. vaex 5.x) when doing such encoding you will continue to see the strings, but under the hood we will treat them as ints. You can get a preview of this feature by doing:

```python
import vaex
df = vaex.datasets.titanic()
df = df._future()  # enables v5 features (preview)
df = df.ordinal_encode('home_dest')
```

**Comment by tommyhj217** [comment]
Hi

Thanks for fast reply. :)
I read your below article and tried to developed dash apps with vaex.
https://medium.com/plotly/interactive-and-scalable-dashboards-with-vaex-and-dash-9b104b2dc9f0

I tested future version ordinal_encode. So  I have one more question.

Do I have to use the dictionary type to use encode column('hode_dest') isin in the future version?

Thanks
Hyunjun.

**Reply by JovanVeljanoski** [reply]
I do not full understand what you mean.. do you want to use the `isin` method with an encoded column? Or something else?

**Reply by tommyhj217** [reply]
Yes. I want to use isin method with an encoded column.
Can't used value_counts with an encoded column?

**Reply by tommyhj217** [reply]
df = vdf.copy()
df = df._future()
df = df.ordinal_encode("course")

df.course.value_counts()

NotImplementedError: Cannot convert dictionary<values=string, indices=int8, ordered=0>

**Reply by JovanVeljanoski** [reply]
That's all to be decided / solved in the future :). This `df._future()` is just a preview and for us to try things out. Feel free to try things out, but no complaining there: those things come without any guarantee and are under development .

**Reply by tommyhj217** [reply]
It's okay if it works later. I tried it because I was curious about how isin and value_counts work.

**Reply by maartenbreddels** [reply]
We didn't merge https://github.com/vaexio/vaex/pull/1841 yet, which should make value_counts work with encoded columns.
